model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleNewDomain_empty() {
+	domain := model.NewDomain()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleSingleton() {
+	domain := model.Singleton(1)
+
+	fmt.Println(domain.Value())
+
+	// Output:
+	// 1 true
+}
+
+func ExampleMultiple_first() {
+	domain := model.Multiple()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleMultiple_second() {
+	domain := model.Multiple(3, 5, 1)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	fmt.Println(domain.Len())
+
+	fmt.Println(domain.Contains(3))
+
+	// Output:
+	// 1 true
+	// 5 true
+	// 3
+	// true
+}
+
+/*
+func ExampleNewDomain_singleRange_first() {
+	domain := model.NewDomain(model.NewRange(1, 0))
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+*/
+
+func ExampleNewDomain_singleRange_second() {
```

Comment:
In my mind, I don't think we need that many examples to understand how to build a new domain using `NewDomain`. I would use the strategy I presented above to maybe have `func ExampleNewDomain_single` and `func ExampleNewDomain_multiple` ➡️ two examples that showcase how to build domains using this function. Inside each function you can place multiple cases and add some comments so that the user follow through. I thikn this makes it more compact for the current example naming is a bit cryptic, in my opinion:

<img width="365" alt="image" src="https://user-images.githubusercontent.com/42215971/187504582-549af73e-8fe2-45b8-a1df-b5c4a1bf1b3e.png">

I also think we are ok with a couple of examples...

Reply:
By the way, the comment here regarding using `first`,  `second` etc. applies here: they don't get rendered in order so it is confusing.